holoviews/core/spaces.py

```diff
@@ -500,6 +517,8 @@ class DynamicMap(HoloMap):
        """)
 
     def __init__(self, callback, initial_items=None, **params):
+        if not isinstance(callback, Callable) and not isinstance(callback, types.GeneratorType):
+            callback = Callable(callable_function=callback)
```

Comment:
For instance, we might want an example of the user declaring a ``Callable`` with their own set of ``inputs``.